src/mm-input-mask/mm-input-mask.js

```diff
@@ -0,0 +1,396 @@
+/**
+ * @license
+ * Copyright (c) 2015 MediaMath Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at http://mediamath.github.io/strand/LICENSE.txt
+
+*/
+(function () {
+
+	_types = {
+		SEP:0,
+		GROUP:1
+	};
+
+	_restrict = {
+		alpha:/[A-Za-z]+/g,
+		numeric:/\d+/ig,
+		alphaNumeric:/[A-Za-z\d]+/g,
+		all:/.+/g
+	};
+
+	_applyReg = function(reg, val) {
+		if (reg) {
+			reg.lastIndex = 0;
+			return reg.test(val);
+		}
+		return false;
+	};
+
+	_rules = {
+		hour12:function(char, input, group, field) {
+
+			var val = input+char,
+				first = val.length === 1,
+				second = val.length === 2,
+				imp = parseInt(char);
+
+			if (first) {
+				if (imp > 1) {
+					field.value = "0" + input;
+					return false;
+				}
+			} else if (second) {
+				if (imp > 2 && parseInt(input) === 1) {
+					field.value = input + "2";
+					return false;
+				}
+			}
+			return true;
+		},
+		hour24: function(char, input, group, field) {
+			var val = input+char,
+				first = val.length === 1,
+				second = val.length === 2,
+				imp = parseInt(char);
+
+		},
+		minutes: function(char, input, group, field) {
+
+			var first = (input+char).length === 1,
+				imp = parseInt(char);
+			if (first) {
+				if (imp > 5) {
+					field.value = "5";
+					return false;
+				}
+			}
+			return true;
+		},
+		range:function(char, input, group, field) {
+			if (input.length < group.max) { return true; }
+			var i = parseInt(input),
+				rng = _parseRange(group.args);
+			return i >= rng[0] && i <= rng[1];
+		},
+		exact: function(char, input, group, field) {
+			return input === group.args;
+		},
+		all: function() {
+			return true;
+		}
+	};
+
+	_parseRange = function(range) {
+		if (!range) return [];
+		return range.split("-").map(function(i) { return parseInt(i); });
+	};
+
+	_groupRange = function(range) {
+		if (!range) return [];
+		if (_isRange(range)) {
+			return _parseRange(range);
+		} else {
+			var s = parseInt(range);
+			return [s,s];
+		}
+	};
+
+	_isRange = function(range) {
+		return range.indexOf("-") !== -1;
+	};
+
+	Polymer('mm-input-mask', {
+		ver:"<<version>>",
+
+		publish: {
+			mask:"",
+			rules:null,
+			restrict:null
+		},
+
+		created: function() {
+			this.loader = new FontLoader();
+			this.loader.add("Arimo").then(function(e) {
+				this.arimoLoaded = true;
+				this.maskConfig.filter(function(group) {
```

Comment:
Seems like it is possible (though perhaps improbable) that this.maskConfig is not assigned by the time this executes (i.e. race condition).

As a preventative measure, it may make sense to store the promise from loader.add() and invoke it's .then() after this.parseMask() in the "ready" lifecycle event (where this.maskConfig is assigned).